driver-core/src/main/com/mongodb/operation/CommitTransactionOperation.java

```diff
@@ -126,19 +141,36 @@ protected String getCommandName() {
 
     @Override
     CommandCreator getCommandCreator() {
-        final CommandCreator creator = super.getCommandCreator();
         if (alreadyCommitted) {
+            final CommandCreator creator = super.getCommandCreator();
             return new CommandCreator() {
                 @Override
                 public BsonDocument create(final ServerDescription serverDescription, final ConnectionDescription connectionDescription) {
                     return getRetryCommandModifier().apply(creator.create(serverDescription, connectionDescription));
                 }
             };
         } else {
-            return creator;
+            return getCommitCommandCreator();
         }
     }
 
+    CommandCreator getCommitCommandCreator() {
```

Comment:
This can be DRY'd up by instead calling `super.getCommandCreator()` and appending the recoveryToken to the `BsonDocument` that its `create` method returns.

Reply:
Done